mitodl/open-discussions
# Move userlist types out of fields, update TS

#### Pre-Flight checklist


#### What are the relevant tickets?
Toward #3819 

#### What's this PR do?
While working on #3819 I came across some issues with our `LearningResource` TS declarations that motivated me to rename a bunch of stuff. This PR includes those renames and a few small other changes. There are no behavioral changes in this PR.

#### How should this be manually tested?
There are no intentional behavioral changes in this PR. The only things that are possibly affected are in Infinite Corridor. You could `docker compose up` and check that:
- http://od.odl.local:8063/infinite still looks OK
- the individual linked field pages still look OK
- http://od.odl.local:8063/infinite/search still looks OK

**To construct field data:** I would do it this way
1. Create some (2+) public userlists at http://od.odl.local:8063/learn (bookmark icon on cards)
2. Create a field via admin panel http://od.odl.local:8063/admin/channels_fields/fieldchannel/
3. Associate userlists with your field via PATCH request at http://od.odl.local:8063/api/v0/fields/YOUR_FIELD_NAME/
    - `featured_list: <list_id>`
    - `subfields: <array of list ids>`

#### Where should the reviewer start?
See my comments on files.
